qiskit_experiments/data_processing/nodes.py

```diff
@@ -848,3 +867,63 @@ def _restless_classify(shot: str, prev_shot: str) -> str:
             restless_adjusted_bits.append("0" if bit == prev_shot[idx] else "1")
 
         return "".join(restless_adjusted_bits)
+
+
+class RestlessToIQ(RestlessNode):
+    """Post-process restless data and convert restless memory to IQ data.
+
+    This node first orders the measured restless IQ point (measurement level 1) data
+    according to the measurement sequence and then subtracts an IQ point from the previous
+    one, i.e. :math:`(I_2 - I_1) + i(Q_2 - Q_1)` for consecutively measured IQ points
+    :math:`I_1 + iQ_1` and :math:`I_2 + iQ_2`. Following this, it takes the absolute
+    value of the in-phase and quadrature component and returns a sequence of circuit-
+    ordered IQ values, e.g. containing :math:`\abs{(I_2 - I_1)} + i\abs{(Q_2 - Q_1)}`.
+    This procedure is based on M. Werninghaus, et al., PRX Quantum 2, 020324 (2021).
+    """
+
+    def __init__(self, validate: bool = True):
+        """
+        Args:
+            validate: If set to False the DataAction will not validate its input.
+        """
+        super().__init__(validate)
+
+    def _process(self, data: np.ndarray) -> np.ndarray:
+        """Reorder the IQ shots and assign values to them based on the previous outcome.
+
+        Args:
+            data: An array representing the memory.
+
+        Returns:
+            An array of arrays of IQ shots processed according to the restless methodology.
+        """
+
+        # Step 1. Reorder the data.
+        memory = self._reorder_iq(data)
+
+        post_processed_memory = [
+            [
+                [np.abs(memory[0][sub_idx][0]), np.abs(memory[0][sub_idx][1])]
+                for sub_idx in range(len(memory[0]))
+            ]
+        ]
+
+        # Step 2. Subtract and take absolute value of consecutive IQ points in
+        # the reordered memory.
+        for idx in range(1, len(memory)):
+            post_processed_memory.append(
+                [
+                    [
+                        np.abs(memory[idx][sub_idx][0] - memory[idx - 1][sub_idx][0]),
```

Comment:
The way the first index, i.e. `idx` is handled seems inconsistent with the first step where it was set to 0.

Reply:
I added some documentation for clarification, see [993a4c6](https://github.com/Qiskit/qiskit-experiments/pull/752/commits/993a4c656fa5e78f569bc070e927beeb982dfd8b).